[CI] QueryableReservedRolesIT testQueryDeleteOrUpdateReservedRoles failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #5877 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/nzqjria6tro6o)
- [elasticsearch-periodic-platform-support #5831 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/umrczgw3pwdug)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:security:qa:security-basic:javaRestTest" --tests "org.elasticsearch.xpack.security.QueryableReservedRolesIT.testQueryDeleteOrUpdateReservedRoles" -Dtests.seed=F21C6BF518702D9D -Dtests.locale=es-PE -Dtests.timezone=Canada/Eastern -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.security.QueryableReservedRolesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testQueryDeleteOrUpdateReservedRoles),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is <29>
     but: was <0>
```

**Issue Reasons:**
- [main] 2 failures in test testQueryDeleteOrUpdateReservedRoles (0.3% fail rate in 636 executions)
- [main] 2 failures in step windows-2022_checkpart4_platform-support-windows (20.0% fail rate in 10 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (20.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryableReservedRolesIT.java
+++ b/x-pack/plugin/security/qa/security-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/QueryableReservedRolesIT.java
@@ -121,14 +121,16 @@ public class QueryableReservedRolesIT extends ESRestTestCase {
         waitForMigrationCompletion(adminClient(), SecurityMigrations.ROLE_METADATA_FLATTENED_MIGRATION_VERSION);
 
         final String[] allReservedRoles = ReservedRolesStore.names().toArray(new String[0]);
-        assertQuery(client(), """
-            { "query": { "bool": { "must": { "term": { "metadata._reserved": true } } } }, "size": 100 }
-            """, allReservedRoles.length, roles -> {
-            assertThat(roles, iterableWithSize(allReservedRoles.length));
-            for (var role : roles) {
-                assertThat((String) role.get("name"), is(oneOf(allReservedRoles)));
-            }
-        });
+        assertBusy(() -> {
+            assertQuery(client(), """
+                { "query": { "bool": { "must": { "term": { "metadata._reserved": true } } } }, "size": 100 }
+                """, allReservedRoles.length, roles -> {
+                assertThat(roles, iterableWithSize(allReservedRoles.length));
+                for (var role : roles) {
+                    assertThat((String) role.get("name"), is(oneOf(allReservedRoles)));
+                }
+            });
+        }, 30, TimeUnit.SECONDS);
 
         final String roleName = randomFrom(allReservedRoles);
         assertQuery(client(), String.format("""

